File: BackEnd/CasottoIdS/src/main/java/it/unicam/cs/CasottoIdS/models/Promozione.java
package it.unicam.cs.CasottoIdS.models;

import org.springframework.data.annotation.Id;

public class Promozione {

    @Id
    public String idPromozione;
    private String idProdotto;
    private double originalPrezzo;
    private double newPrezzo;

    public Promozione(String idProdotto, double originalPrezzo, double newPrezzo) {
        this.idProdotto = idProdotto;
        this.originalPrezzo = originalPrezzo;
        this.newPrezzo = newPrezzo;
    }

    public String getIdPromozione() {
        return idPromozione;
    }

   public String getIdProdotto() {
        return idProdotto;
    }

    public void setIdProdotto(String idProdotto) {
        this.idProdotto = idProdotto;
    }

    public double getOriginalPrezzo() {
        return originalPrezzo;
    }

    public void setOriginalPrezzo(double originalPrezzo) {
        this.originalPrezzo = originalPrezzo;
    }

    public double getNewPrezzo() {
        return newPrezzo;
    }

    public void setNewPrezzo(double newPrezzo) {
        this.newPrezzo = newPrezzo;
    }
}

File: FrontEnd/Casotto/lib/models/Promozione.dart
class Promozione {
  late String idPromozione;
  late String idProdotto;
  late double originalPrezzo;
  late double newPrezzo;

  Promozione(String idPromozione, String idProdotto, double originalPrezzo,
      double newPrezzo) {
    this.idPromozione = idPromozione;
    this.idProdotto = idProdotto;
    this.originalPrezzo = originalPrezzo;
    this.newPrezzo = newPrezzo;
  }

  String getIdPromozione() {
    return this.idPromozione;
  }

  String getIdProdotto() {
    return this.idProdotto;
  }

  void setIdProdotto(String idProdotto) {
    this.idProdotto = idProdotto;
  }

  double getOriginalPrezzo() {
    return this.originalPrezzo;
  }

  void setOriginalPrezzo(double originalPrezzo) {
    this.originalPrezzo = originalPrezzo;
  }

  double getNewPrezzo() {
    return this.newPrezzo;
  }

  void setNewPrezzo(double newPrezzo) {
    this.newPrezzo = newPrezzo;
  }
}
